Source/Charts/Renderers/LineChartRenderer.swift

```diff
@@ -751,4 +767,113 @@ open class LineChartRenderer: LineRadarRenderer
         context.restoreGState()
     }
+
+    /// Generates the path that is used for gradient drawing.
+    private func generateGradientLinePath(dataSet: LineChartDataSetProtocol, fillMin: CGFloat, from: Int, to: Int, matrix: CGAffineTransform) -> CGPath
+    {
+        let phaseX = CGFloat(animator.phaseX)
+        let phaseY = CGFloat(animator.phaseY)
+
+        var e: ChartDataEntry!
+
+        let generatedPath = CGMutablePath()
+        e = dataSet.entryForIndex(from)
+        if e != nil
+        {
+            generatedPath.move(to: CGPoint(x: CGFloat(e.x), y: CGFloat(e.y) * phaseY), transform: matrix)
+        }
+
+        // create a new path
+        let to = Int(ceil(CGFloat(to - from) * phaseX + CGFloat(from)))
+        for i in (from + 1)..<to
+        {
+            guard let e = dataSet.entryForIndex(i) else { continue }
+            generatedPath.addLine(to: CGPoint(x: CGFloat(e.x), y: CGFloat(e.y) * phaseY), transform: matrix)
+        }
+        return generatedPath
+    }
+
+    func drawGradientLine(context: CGContext, dataSet: LineChartDataSetProtocol, spline: CGPath, matrix: CGAffineTransform)
+    {
+        context.saveGState()
+        defer { context.restoreGState() }
+
+        let gradientPath = spline.copy(strokingWithWidth: dataSet.lineWidth, lineCap: .butt, lineJoin: .miter, miterLimit: 10)
+        context.addPath(gradientPath)
+        context.drawPath(using: .fill)
+
+        let boundingBox = gradientPath.boundingBox
+        let gradientStart = CGPoint(x: 0, y: boundingBox.maxY)
+        let gradientEnd = CGPoint(x: 0, y: boundingBox.minY)
+        var gradientLocations = [CGFloat]()
+        var gradientColors = [CGFloat]()
+        var cRed: CGFloat = 0
+        var cGreen: CGFloat = 0
+        var cBlue: CGFloat = 0
+        var cAlpha: CGFloat = 0
+
+        //Set lower bound color
+        gradientLocations.append(0)
+        var cColor = dataSet.color(atIndex: 0)
+        if cColor.getRed(&cRed, green: &cGreen, blue: &cBlue, alpha: &cAlpha)
+        {
+            gradientColors += [cRed, cGreen, cBlue, cAlpha]
+        }
+
+        //Set middle colors
+        guard let gradientPositions = dataSet.gradientPositions else
+        {
+            fatalError("Must set `gradientPositions if `dataSet.isDrawLineWithGradientEnabled` is true")
+        }
+
+        for position in gradientPositions
+        {
+            let positionLocation = CGPoint(x: 0, y: position)
+                .applying(matrix)
+            let normPositionLocation = (positionLocation.y - gradientStart.y) / (gradientEnd.y - gradientStart.y)
+            if (normPositionLocation < 0) {
+                gradientLocations.append(0)
+            } else if (normPositionLocation > 1) {
+                gradientLocations.append(1)
+            } else {
+                gradientLocations.append(normPositionLocation)
+            }
+        }
+
+        for _ in dataSet.colors
+        {
+            cColor = dataSet.color(atIndex: 0)
```

Comment:
@jjatie just want to check that this line is correct. It iterates over `dataSet.colors`, but always takes color at index `0`.

Reply:
for i in dataSet.colors.indices
        {
            cColor = dataSet.color(atIndex: i)